### PR TITLE
Add benchmarking example for ipc functions

### PR DESCRIPTION
**Pull request by rakeshsadhu**

### CODE REVIEW

File: tests/benchmarking/tst_benchmarking_inprocess.qml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    BenchMarkingIPCProxy {
+        id: api
+        ipc.objectPath: "/benchmark/models/inprocess"
+    }
+
+
+    function initTestCase() {
+        Benchmark.checkInit();
```
Comment (jacky309) [inline tests/benchmarking/tst_benchmarking_inprocess.qml]: BenchMark, Benchmark, Benchmarking ?
Please make up your mind and be consistent

File: tests/benchmarking/BenchMarkingImplementation.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    {
+        return intData;
+    }
+
+    QString sendString(QString someString)
+    {
+        return someString;
+    }
+
+    facelift::example::benchmarking::LargeStruct sendStructure(facelift::example::benchmarking::LargeStruct structData) override
```
Comment (jacky309) [inline tests/benchmarking/BenchMarkingImplementation.h]: this method does not seem to send anything

File: tests/benchmarking/BenchMarkingImplementation.h [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    SmallStruct smallModelData(int)
+    {
+        SmallStruct entry;
+        entry.setid(42);
+        return entry;
+    }
+
+    LargeStruct largelModelData(int)
+    {
+        static const QStringList sl {QStringLiteral("I stand amid the roar"),
```
Comment (jacky309) [inline tests/benchmarking/BenchMarkingImplementation.h]: align

File: tests/benchmarking/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    INTERFACE_DEFINITION_FOLDER ${CMAKE_CURRENT_SOURCE_DIR}/interface
+)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+set(CMAKE_AUTOMOC ON)
+
+facelift_add_qml_plugin(BenchMarkingPlugin
+    URI tests.benchmarking
+    OUTPUT_BASE_DIRECTORY imports/cpp
+    SOURCES_GLOB_RECURSE ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp
```
Comment (jacky309) [inline tests/benchmarking/CMakeLists.txt]: no glob

File: tests/benchmarking/BenchMarkingPlugin.cpp [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#include "facelift/example/benchmarking/Module.h"
+#include "BenchMarkingImplementation.h"
+
+using namespace facelift::example::benchmarking;
+
+void BenchMarkingPlugin::registerTypes(const char *uri)
+{
+    Module::registerQmlTypes(uri);
+    Module::registerUncreatableQmlTypes(uri);
+    facelift::registerQmlComponent<BenchmarkingImplementation> (uri , "BenchMarkingInterface");
```
Comment (jacky309) [inline tests/benchmarking/BenchMarkingPlugin.cpp]: Why "BenchMarkingInterface" now ?

**Comment by rakeshsadhu** [comment]
@jacky309  please do review

File: tests/benchmarking/BenchMarkingImplementation.h [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        return someString;
+    }
+
+    facelift::example::benchmarking::LargeStruct sendStructure(facelift::example::benchmarking::LargeStruct structData) override
+    {
+        return structData;
+    }
+
+    facelift::example::benchmarking::EnumType sendEnum(facelift::example::benchmarking::EnumType eType) override
```
Comment (jacky309) [inline tests/benchmarking/BenchMarkingImplementation.h]: same

File: tests/benchmarking/BenchMarkingImplementation.h [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        return eType;
+    }
+
+
+    void sendIntegerAsync(int intData, facelift::AsyncAnswer<int> answer) override
```
Comment (jacky309) [inline tests/benchmarking/BenchMarkingImplementation.h]: requestInteger

File: tests/benchmarking/interface/BenchMarking.qface [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+struct LargeStruct {
+    int id
+    string name
+    list<string> texts
+    EnumType type
+}
+
+enum EnumType {
```
Comment (jacky309) [inline tests/benchmarking/interface/BenchMarking.qface]: Type ?

File: tests/benchmarking/tst_benchmarking_inprocess.qml [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+import QtTest 1.2
+import tests.benchmarking 1.0
+import "BenchmarkingHelper.js" as JsBenchmarking
+
+TestCase {
+    name: "benchmark-models-inprocess"
+
+    BenchMarking {
+        IPC.enabled: true
+        IPC.objectPath: "/benchmark/models/inprocess"
```
Comment (jacky309) [inline tests/benchmarking/tst_benchmarking_inprocess.qml]: remove

File: tests/benchmarking/tst_benchmarking_inprocess.qml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+    }
+
+    BenchMarkingIPCProxy {
+        id: api
+        ipc.objectPath: "/benchmark/models/inprocess"
```
Comment (jacky309) [inline tests/benchmarking/tst_benchmarking_inprocess.qml]: remove

**Pull request merged**